### PR TITLE
Add support for empty dicts and lists

### PR DESCRIPTION
**Pull request by kulheim**
### Summary
Support for empty dicts and lists, see new test.

### CODE REVIEW

**Comment by amirziai** [comment]
@kulheim thanks for the PR. Seems like the branch has conflicts with master, which I think was introduced from a very recent PR. Do you mind pulling the latest and resolving the conflict?

**Comment by kulheim** [comment]
Conflict resolved.

**Comment by amirziai** [comment]
@kulheim thank you! Travis is failing to build with status `Abuse Detected`. I sent an email to support, will update you when I hear back.

**Comment by amirziai** [comment]
@kulheim Travis wrote back to me:
```
We've been experiencing some issues lately where, if a pull request is opened from an organization that has never used Travis creates a pull request on a repo that does, our system is automatically rejecting it as abuse. While we work on resolving this issue, if you can have the user who opened the pull request log into travis-ci.org and sync their Github account, I'll be able to go in and manually remove the abusive flag and it should allow the PR to build.
```

Do you mind linking your GitHub account to Travis? Sorry about this.

**Comment by kulheim** [comment]
@amirziai gh account linked to Travis, sorry for inconvenience.

**Comment by amirziai** [comment]
@kulheim thank you. I asked Travis to resolve the issue.

File: flatten_json.py [hunk 1]
```diff
@@ -46,7 +46,9 @@ def _flatten(object_, key):
         :param key: carries the concatenated key for the object_
         :return: None
         """
-        if isinstance(object_, dict):
+        if not object_:
```
Comment (amirziai) [inline flatten_json.py]: This is great. Can you add a comment here describing why this is necessary for future reference please?

File: test_flatten.py [hunk 2]
```diff
@@ -97,6 +97,16 @@ def test_list_and_dict(self):
         actual = flatten(dic)
         self.assertEqual(actual, expected)
 
+    def test_empty_list_and_dict(self):
+        dic = {
+            'a': {},
```
Comment (amirziai) [inline test_flatten.py]: Probably obvious but can you add keys with `None` and `""` values that will also be caught correctly by your modification? Can't think of any other cases but if you can it'd be great to add them.

**Comment by kulheim** [comment]
Comments and test added.

**Comment by amirziai** [comment]
thanks @kulheim. merged!

**Pull request merged**